pyre/zbeacon.py

```diff
@@ -137,57 +136,78 @@ def prepare_udp(self):
-                logger.debug("Checking out interface {0}.".format(name))
-                # For some reason the data we need lives in the "2" section of the interface.
-                data_2 = data.get(2)
+            if not data_2:
+                logger.debug("No data_2 found for interface {0}.".format(name))
+                return
 
-                if not data_2:
-                    logger.debug("No data_2 found for interface {0}.".format(name))
-                    continue
+            address_str = data_2.get("addr")
+            netmask_str = data_2.get("netmask")
 
-                address_str = data_2.get("addr")
-                netmask_str = data_2.get("netmask")
+            if not address_str or not netmask_str:
+                logger.debug("Address or netmask not found for interface {0}.".format(name))
+                return
 
-                if not address_str or not netmask_str:
-                    logger.debug("Address or netmask not found for interface {0}.".format(name))
-                    continue
+            if isinstance(address_str, bytes):
+                address_str = address_str.decode("utf8")
 
-                if isinstance(address_str, bytes):
-                    address_str = address_str.decode("utf8")
+            if isinstance(netmask_str, bytes):
+                netmask_str = netmask_str.decode("utf8")
 
-                if isinstance(netmask_str, bytes):
-                    netmask_str = netmask_str.decode("utf8")
+            interface_string = "{0}/{1}".format(address_str, netmask_str)
 
-                interface_string = "{0}/{1}".format(address_str, netmask_str)
+            interface = ipaddress.ip_interface(u(interface_string))
 
-                interface = ipaddress.ip_interface(u(interface_string))
+            if interface.is_loopback:
+                logger.debug("Interface {0} is a loopback device.".format(name))
+                return
 
-                if interface.is_loopback:
-                    logger.debug("Interface {0} is a loopback device.".format(name))
-                    continue
+            if interface.is_link_local:
+                logger.debug("Interface {0} is a link-local device.".format(name))
+                return
 
-                if interface.is_link_local:
-                    logger.debug("Interface {0} is a link-local device.".format(name))
-                    continue
-
-                self.address = interface.ip
-                self.network_address = interface.network.network_address
-                self.broadcast_address = interface.network.broadcast_address
-                self.interface_name = name
+            self.address = interface.ip
+            self.network_address = interface.network.network_address
+            self.broadcast_address = interface.network.broadcast_address
+            self.interface_name = name
 
             if self.address:
-                break
+                return
```

Comment:
I think this is fine as a first implementation. Have you looked at how this is handled in zyre(czmq). The interface is selected by either its name or its address. In case the string is a single digit it just uses the index in the interface list.